src/imitation/policies/exploration_wrapper.py

```diff
@@ -64,8 +71,20 @@ def _switch(self) -> None:
         else:
             self.current_policy = self.wrapped_policy
 
-    def __call__(self, obs: np.ndarray) -> np.ndarray:
-        acts = self.current_policy(obs)
+    def __call__(
+        self,
+        observation: np.ndarray,
+        state: Optional[Tuple[np.ndarray, ...]],
+        _episode_start: Optional[np.ndarray],
+    ) -> Tuple[np.ndarray, Optional[Tuple[np.ndarray, ...]]]:
+        if state is not None:
+            raise ValueError("Exploration wrapper does not support stateful policies.")
+
+        acts, state = self.current_policy(observation, None, None)
+
+        if state is not None:
```

Comment:
Do we need both of these checks?

Reply:
I guess technically you could only use one, but they are checking different things. Probably the check for the return value is more critical. But I'll add comments explaining the rationale for having both for now

Reply:
It definitely *looks* redundant but checking both input and return value will catch errors faster and more localized (although I struggle to think of cases where an error wouldn't be caught on the next step even if you omitted one check). Adding comments to justify this sounds good. You could also consider modifying the error message a bit between them to emphasize where the non-None state is coming from.